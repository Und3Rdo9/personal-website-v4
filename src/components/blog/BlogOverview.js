import React from "react";
import Container from './../common/Container';
import Section from './../common/Section';
import SectionTitle from './../common/SectionTitle';
import AllPostsFeedContainer from './AllPostsFeedContainer';

const BlogOverview = () => (
  <Container>
    <Section>
      <SectionTitle title='Blogartikelen' />
      <AllPostsFeedContainer />
    </Section>
  </Container>
);

export default BlogOverview;
