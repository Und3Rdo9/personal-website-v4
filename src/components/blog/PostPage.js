import React from 'react';
import { gql } from 'apollo-boost';
import { Query } from 'react-apollo';
import ReactMarkdown from 'react-markdown';
import SectionLoader from './../common/SectionLoader';

const GET_POST = gql`
query post($slug: String!) {
  Post(slug: $slug) {
    title
    authors {
      name
    }
    dateAndTime
    summary
    tags
    content
  }
}
`;

const PostPage = ( {match} ) => (
  <div>
    <Query query={GET_POST} variables={{ slug: match.params.slug }}>
      {( { loading, error, data } ) => {
        if (loading) return <SectionLoader isActive={true} />;
        if (error) return <div>Error :(</div>;
          if (data.Post) {
            return (
              <article className="container">
                <h1>{data.Post.title}</h1>
                <p>{`by ${data.Post.authors[0].name}`}</p>
                <p>{data.Post.postsummary}</p>
                <main>
                  <ReactMarkdown source={data.Post.content} />
                </main>
              </article>
            )
          }
          else {
            return (
              <article>
                <p>Sorry, this post has not been found.</p>
              </article>
            )
          }

        }}
      </Query>
  </div>
)

export default PostPage;
