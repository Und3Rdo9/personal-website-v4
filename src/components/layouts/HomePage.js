import React, { Component } from 'react';
import Hero from './../hero/Hero';
import UspListContainer from './../usp/UspListContainer';
import PostFeedContainer from './../blog/PostFeedContainer';
import portraitImg from './../../assets/images/portrait-tb.jpeg';
import Section from './../common/Section';
import Container from './../common/Container';

class HomePage extends Component {
  render() {
    return (
      <React.Fragment>
        <Container>
          <Hero
            title='Tim Bakkum'
            underline='Amsterdam based front&#8209;end&nbsp;developer'
            imageSrc={portraitImg}
          />
        </Container>

        <Section type='light'>
          <h2 style={{textAlign:'center'}}>What I'm About</h2>
          <Container>
            <UspListContainer />
          </Container>
        </Section>

        <Container>
          <h2 style={{textAlign:'center'}}>Latest from my front&#8209;end blog</h2>
          <PostFeedContainer />
        </Container>
      </React.Fragment>

    );
  }
}

export default HomePage;
