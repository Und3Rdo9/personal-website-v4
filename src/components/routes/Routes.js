import React from 'react';
import { Route, Switch } from 'react-router-dom';

import HomePage from './../layouts/HomePage';
import PostPageContainer from './../blog/PostPageContainer';
import ContentPageContainer from './../page/ContentPageContainer';
import BlogOverview from './../blog/BlogOverview';

const Routes = () => (
  <main>
    <Switch>
      <Route exact path='/' component={HomePage} />
      <Route exact path='/frontend-blog' component={BlogOverview} />
      <Route exact path='/frontend-blog/:slug' component={PostPageContainer} />
      <Route exact path='/amsterdam-frontend-developer/:slug' component={ContentPageContainer} />
    </Switch>
  </main>
);

export default Routes;
