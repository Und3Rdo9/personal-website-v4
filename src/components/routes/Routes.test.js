import React from 'react';
import { mount } from 'enzyme';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { MockedProvider } from 'react-apollo/test-utils';
import { MemoryRouter } from 'react-router';

import HomePage from './../layouts/HomePage';
import BlogOverview from './../blog/BlogOverview';
import PostPage from './../blog/PostPage';

import Routes from './Routes';

configure({ adapter : new Adapter() });

describe('<Routes />', () => {

  it('renders the HomePage component correctly when visiting / ', () => {
    const router = mount(
      <MockedProvider mocks={[]}>
        <MemoryRouter initialEntries={['/']} initialIndex={0} >
          <Routes />
        </MemoryRouter>
      </MockedProvider>
    );

    expect(router.find(HomePage).length).toBe(1);
  });

  it('renders the matched route component correctly', () => {
    const router = mount(
      <MemoryRouter initialEntries={['/frontend-blog']} initialIndex={0} >
        <Routes />
      </MemoryRouter>
    );

    expect(router.find(BlogOverview).length).toBe(1);
  });

  it('renders a blog post page component correctly', () => {
    const router = mount(
      <MockedProvider mocks={[]}>
        <MemoryRouter initialEntries={['/frontend-blog/my-javascript-journey']} initialIndex={0} >
          <Routes />
        </MemoryRouter>
      </MockedProvider>
    );

    expect(router.find(PostPage).length).toBe(1);
  });
});
