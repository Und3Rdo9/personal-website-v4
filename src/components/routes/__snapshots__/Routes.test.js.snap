// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes /> matches the snapshot 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "3je6gh",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "3je6gh",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Routes />
</Router>
`;
